buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -297,10 +298,12 @@ class BeamModulePlugin implements Plugin<Project> {
     // Provide code coverage
     // TODO: Should this only apply to Java projects?
     project.apply plugin: "jacoco"
+
     project.gradle.taskGraph.whenReady { graph ->
       // Disable jacoco unless report requested such that task outputs can be properly cached.
       // https://discuss.gradle.org/t/do-not-cache-if-condition-matched-jacoco-agent-configured-with-append-true-satisfied/23504
       def enabled = graph.allTasks.any { it instanceof JacocoReport }
+      enabled = enabled || graph.allTasks.any { it.name.contains("sonarqube") }
```

Comment:
Do you need to attach some dependsOn relationship here? Our JacocoReport task is broken because of this - you have to say `gradlew test jacocoReport` when that should be automatic and it should just be `gradlew jacocoReport`.

Reply:
imho, this should not be necessary. AFAICT, there are already multiple task setup. Why not just add jacocoTestReport to that list and u are fine?